src/util/Concepts.h

```diff
@@ -13,4 +13,7 @@ namespace ad_utility {
  */
 template <typename T>
 concept Streamable = requires(T x, std::ostream& os) { os << x; };
+
+template <typename Func, typename R, typename... ArgTypes>
+concept InvocableWithReturnValue = std::is_invocable_r_v<R, Func, ArgTypes...>;
```

Comment:
1. needs a comment (The use of it is that there is no concept for `std::is_invocable_r_v`.
2. We have additional concepts in the mean time that do similar things, but prohibit conversions on the return type (`InvokableWithExactReturnType` and `InvocableWithSimilarReturnType`. This concept should be defined close to those and then called `InvocableWithConvertibleReturnType`.